### DISCUSSION TITLE
How I can use this data with API?

### DISCUSSION THREAD

**Discussion by mbhusty** [f1db/f1db#31] [Q&A] [answered]
Hi! I build my own app on React Native with F1 data. How I can use this data by API request? I'm sorry for the stupid question)

**Comment by marceloverdijk** [comment]
Hi @mbhusty there are no stupid questions and thx for reaching out.

F1DB does not offer a API (yet).
It's on my TODO list and definitely something which will be available in the future.

For now you could use de json or SQLite database to build your own API. Does that help?

**Reply by mbhusty** [reply]
> json or SQLite database

Can u explain me how I can use this options? how I can connect to SQLite database?

**Reply by marceloverdijk** [reply]
@mbhusty Maybe this will give you a start: https://github.com/f1db/f1db/discussions/30
This is specifically for Python, so I don't know which programming language you want to use..

**Reply by mbhusty** [reply]
> which programming language

React Native (typescript) without any backend. I think I need server now) So, how I can use data from this repo? 
I can hardly make such a request like that) https://raw.githubusercontent.com/f1db/f1db/main/src/data/seasons/2024/races/01-bahrain/driver-standings.yml

**Reply by marceloverdijk** [reply]
No you are pointing to the source files that are used to generate the output (json, csv, sqlite). 
You should download a release from the [releases](https://github.com/f1db/f1db/releases) page; so either json, smile, csv or sqlite.
And based on which format you choose you should build your own implementation.

As F1DB does not provide a API yet, you have several options:

- When building a static website (e.g. using Next.js, Astro, ...) query the SQLite database during generation using plain SQL or an ORM like [Prisma](https://www.prisma.io/). Note that's what I'm currently working on for F1DB.com itself.
- When you want a API, setup a server and provide endpoints for your needed data. The data can be read and cached either from the csv/json files or from the SQLite database. But this all depends on what you are trying to accomplish, how many users and how many concurrent requests.

**Reply by mbhusty** [reply]
Do I understand correctly that with a new release, I will need to manually update the database?

**Reply by marceloverdijk** [reply]
No, with every new release (after each rave) you just download the updated SQLite database or json/csv files from the [releases](https://github.com/f1db/f1db/releases) page. 
Then you just incorporate them in your project, depending on how you use them.
There is no need to build/update the database yourself.

**Reply by mbhusty** [reply]
I'm sorry, my English is probably not very good. I'll try to explain in more detail)

1. Download the release file, for example json
2. I place this file on my server
3. After the release of a new release, download a new file and replace it on the server? Any variants for automatic updates?

**Reply by marceloverdijk** [reply, marked as answer]
Yes, that could be your workflow.
You either restart the server, or your server code is implemented in such a way that it picks up the file when updated (timestamp?).
For automatic updates you will need some kind of process to check if there is a new release, download the release and copy it to your server.

**Reply by mbhusty** [reply]
Thnx! I think I could do this)
